dump/src/main.rs

```diff
@@ -46,36 +46,39 @@ fn main() {
     });
 
     let App {
-        file: filename,
+        files: filenames,
         no_text_limit,
         width,
         color,
     } = App::from_args();
 
-    let obj = open_file(filename).unwrap_or_else(|e| {
-        report(e);
-        std::process::exit(ERROR_READ);
-    });
-
     let width = width
         .or_else(|| term_size::dimensions().map(|(width, _)| width as u32))
         .unwrap_or(120);
 
     let mut options = DumpOptions::new();
-    match options
-        .no_text_limit(no_text_limit)
-        .width(width)
-        .color_mode(color)
-        .dump_file(&obj)
-    {
-        Err(ref e) if e.kind() == ErrorKind::BrokenPipe => {
-            // handle broken pipe separately with a no-op
-        }
-        Err(e) => {
-            eprintln!("[ERROR] {}", e);
-            std::process::exit(ERROR_PRINT);
+
+    for filename in filenames {
+        let obj = open_file(filename).unwrap_or_else(|e| {
+            report(e);
+            std::process::exit(ERROR_READ);
```

Comment:
This `exit` here and the one in line 78 mean that the program is abruptly terminated if any of the files could not be opened, but there could be others on the list that could be dumped without issue.

I would propose only exiting immediately upon a new flag, something like `--fail-first`, otherwise it would accumulate the number of failed dump attempts and send an exit code based on that. But this is OK to be done in a separate PR if you wish.

Reply:
I've updated this so that exiting immediately is optional, using the suggested flag `--fail-first` . Errors are being accumulated but are not currently used to send an exit code. Do you have a suggestion/preference for exit code logic?

Reply:
Good question. The current error code logic makes sense for single file dumping, but not for multiple files. So my idea is: if the user passed more than one file, let's make it so that the exit code is the number of files which failed to dump successfully (so if 2 files failed, exit with `2`). To the best of my knowledge, this is the behavior of DCMTK's dcmdump. It also doesn't collide with the single-file error codes because negative integers were used there.

Reply:
That sounds great to me. I've pushed another commit to return these exit codes now.